scripts/build-export.ts

```diff
@@ -0,0 +1,113 @@
+import changeCase from 'change-case'
+import fs from 'fs/promises'
+import glob from 'glob'
+import path from 'path'
+import { transform } from '@svgr/core'
+import { transformSync } from 'esbuild'
+
+const DIRECTORY_SOURCE = './src'
+const DIRECTORY_OUTPUT = './dist'
+
+const optimisedComponentTemplate = (
+  { props, imports, componentName, jsx },
+  { tpl }
+) => tpl`${imports};export const ${componentName} = (${props}) => ${jsx}`
+
+const svgoConfig = {
+  multipass: true,
+  plugins: [
+    {
+      name: 'preset-default',
+      params: {
+        overrides: {
+          removeViewBox: false,
+          removeUselessStrokeAndFill: false
+        }
+      }
+    },
+    { name: 'removeAttrs', params: { attrs: 'aria-labelledby' } },
+    { name: 'removeDimensions' },
+    { name: 'removeTitle' }
+  ]
+}
+
+const exists = async (path: string) => {
+  try {
+    await fs.access(path)
+    return true
+  } catch {
+    return false
+  }
+}
+
+const createESExportString = async (icons: Record<string, string>) => {
+  // access all svg file paths from location
+  // const filePaths = await glob.sync(`${source}/*.svg`)
+  // apply export template to each and combine into single string
+  return Object.entries(icons)
+    .map(([name]) => `export * from './${name}'\n`)
+    .join('')
+}
+
+const getIconExports = async (
+  source: string
+): Promise<Record<string, string>> => {
+  const filePaths = await glob.sync(`${source}/*.svg`)
+
+  return filePaths.reduce(
+    (collection, filePath) => ({
+      ...collection,
+      [path.basename(filePath, path.extname(filePath))]: filePath
+    }),
+    {}
+  )
+}
+
+const transformIcon = async (name: string, source: string) => {
+  try {
+    const svgSource = await fs.readFile(source, { encoding: 'utf8' })
+
+    const svgAsComponent = await transform(
+      svgSource,
+      {
+        plugins: ['@svgr/plugin-svgo', '@svgr/plugin-jsx'],
+        svgoConfig,
+        template: optimisedComponentTemplate
+      },
+      { componentName: changeCase.pascalCase(name) }
+    )
+
+    return await transformSync(svgAsComponent, {
+      loader: 'jsx',
```

Comment:
does this need to include tsx?

Reply:
Nope, this only runs on the code returned from the `@svgr/core` transform method, which will always return JSX.